apps/frontend/src/lib/storage/localStorage/user/connection/message/index.ts

```diff
@@ -0,0 +1,151 @@
+import { CreateBackupData, MessageData } from "@types";
+import {
+  decryptReceivedMessage,
+  parseSerializedTapBackMessage,
+  TapBackMessage,
+} from "@/lib/message";
+import { saveBackupAndUpdateStorage } from "../../../utils";
+import { getUserAndSession } from "../..";
+import { Connection } from "@/lib/storage/types";
+import {
+  createActivityBackup,
+  createConnectionBackup,
+  createLastMessageFetchedAtBackup,
+} from "@/lib/backup";
+import { createTapBackReceivedActivity } from "@/lib/activity";
+
+export const processNewMessages = async (
+  messages: MessageData[]
+): Promise<void> => {
+  if (messages.length === 0) {
+    return;
+  }
+
+  const { user, session } = await getUserAndSession();
+
+  const connectionTapBacks: Record<string, TapBackMessage[]> = {};
+
+  let lastMessageTimestamp: Date | null = null;
+  for (const message of messages) {
+    // Ensure messages are in order and update lastMessageTimestamp
+    if (lastMessageTimestamp && message.createdAt < lastMessageTimestamp) {
+      throw new Error("Messages are not in order");
```

Comment:
I think this error would be surfaced to the user in `MyApp`.

Reply:
If messages are out of order how can / should we resolve the issue? Would it be better to sort the messages before the loop and / or return an ordering warning?

Reply:
good catch, sorted before loop